paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -489,18 +498,36 @@ public List<Changelog> safelyGetAllChangelogs() throws IOException {
                         .map(id -> longLivedChangelogPath(id))
                         .collect(Collectors.toList());
 
-        List<Changelog> changelogs = new ArrayList<>();
-        for (Path path : paths) {
-            try {
-                String json = fileIO.readFileUtf8(path);
-                changelogs.add(Changelog.fromJson(json));
-            } catch (FileNotFoundException ignored) {
-            }
-        }
+        List<Changelog> changelogs = Collections.synchronizedList(new ArrayList<>());
+        collectSnapshots(
+                path -> {
+                    try {
+                        changelogs.add(Changelog.fromJson(fileIO.readFileUtf8(path)));
+                    } catch (IOException e) {
+                        if (!(e instanceof FileNotFoundException)) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                },
+                paths);
 
         return changelogs;
     }
 
+    private void collectSnapshots(Consumer<Path> pathConsumer, List<Path> paths)
+            throws IOException {
+        ExecutorService executor =
```

Comment:
Why not use ThreadPoolUtils#createCachedThreadPool？

Reply:
fix